### ISSUE TITLE
`create_resource` fails to deserialize response

### ISSUE THREAD

**Issue by 0rphon** [fastly/fastly-rust#9]
**Version**

fastly-api = "4.5.0"

**What happened**

`fastly_api::apis::resource_api::create_resource` seems to fail to deserialize the response. The following example successfully links the resource on fastly's end, but returns a serde error instead of a response.

### Example
```rust
use fastly_api::apis::{
    configuration::{ApiKey, Configuration},
    resource_api::{create_resource, CreateResourceParams},
};

#[tokio::main]
async fn main() {
    let mut config = Configuration {
        api_key: Some(ApiKey {
            prefix: None,
            key: "MY_API_KEY".into(), // placeholder
        }),
        ..Default::default()
    };
    create_resource(
        &mut config,
        CreateResourceParams {
            service_id: "MY_SERVICE".into(), // placeholder
            version_id: 1,
            resource_id: Some("MY_RESOURCE"into()), // placeholder
            name: Some("my_config".into()),
        },
    )
    .await
    .unwrap();
}
```

### Error
```rust
thread 'main' panicked at src/main.rs:25:6:
called `Result::unwrap()` on an `Err` value: Serde(Error("unknown variant `config`, expected one of `object-store`, `secret-store`, `config-store`", line: 1, column: 262))
stack backtrace:
   0: rust_begin_unwind
             at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/std/src/panicking.rs:645:5
   1: core::panicking::panic_fmt
             at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/core/src/panicking.rs:72:14
   2: core::result::unwrap_failed
             at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/core/src/result.rs:1654:5
   3: core::result::Result<T,E>::unwrap
             at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/core/src/result.rs:1077:23
   4: playground::main::{{closure}}
             at ./src/main.rs:15:5
   5: tokio::runtime::park::CachedParkThread::block_on::{{closure}}
             at /home/orphon/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.37.0/src/runtime/park.rs:281:63
   6: tokio::runtime::coop::with_budget
             at /home/orphon/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.37.0/src/runtime/coop.rs:107:5
   7: tokio::runtime::coop::budget
             at /home/orphon/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.37.0/src/runtime/coop.rs:73:5
   8: tokio::runtime::park::CachedParkThread::block_on
             at /home/orphon/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.37.0/src/runtime/park.rs:281:31
   9: tokio::runtime::context::blocking::BlockingRegionGuard::block_on
             at /home/orphon/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.37.0/src/runtime/context/blocking.rs:66:9
  10: tokio::runtime::scheduler::multi_thread::MultiThread::block_on::{{closure}}
             at /home/orphon/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.37.0/src/runtime/scheduler/multi_thread/mod.rs:87:13
  11: tokio::runtime::context::runtime::enter_runtime
             at /home/orphon/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.37.0/src/runtime/context/runtime.rs:65:16
  12: tokio::runtime::scheduler::multi_thread::MultiThread::block_on
             at /home/orphon/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.37.0/src/runtime/scheduler/multi_thread/mod.rs:86:9
  13: tokio::runtime::runtime::Runtime::block_on
             at /home/orphon/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.37.0/src/runtime/runtime.rs:351:45
  14: playground::main
             at ./src/main.rs:15:5
  15: core::ops::function::FnOnce::call_once
             at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/core/src/ops/function.rs:250:5
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

**Comment by 0rphon** [comment]
the same is true for `fastly_api::apis::resource_api::list_resources`, and probably every other API call that returns a `fastly_api::models::resource_response::ResourceResponse`

**Comment by Integralist** [comment]
👋🏻 Thanks @0rphon for raising this issue.

@kpfleming I think there is a discrepancy between this API client code and the source OpenAPI schema.

If we look at the [API docs](https://www.fastly.com/documentation/reference/api/services/resource/) (built from our OpenAPI specs) we see...

> resource_type: One of: `kv-store`, `secret-store`, `config`

Which aligns with the actual API response. See also the reported error message: `unknown variant 'config'`, which aligns with our OpenAPI schema...

```yaml
type_resource:
  type: string
  description: Resource type
  enum:
    - kv-store
    - secret-store
    - config
```

But the fastly-api crate (this repo, also built from our OpenAPI specs) _doesn't_ contain that. Instead, it has...

https://github.com/fastly/fastly-rust/blob/66e1eda7791c78f6e047266ac58deb2291061950/src/models/type_resource.rs#L10-L20

**UPDATE:** OK, looks like @harmony7 updated the OpenAPI schema last week and changed it from `config-store` to `config` so I think we're just waiting for a new API client release to be published.

**Comment by harmony7** [comment]
Released v4.6.0 crate, which corrects this:

https://github.com/fastly/fastly-rust/blob/cfa52f80058154f8a678a73409d4356b52a84e9b/src/models/type_resource.rs#L10-L20

This should fix the issue, but please feel free to reopen if you still have issues.
